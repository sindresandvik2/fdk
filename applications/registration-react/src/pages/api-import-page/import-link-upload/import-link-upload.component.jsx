import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Button } from 'reactstrap';
import { withState, withHandlers, compose } from 'recompose';

import localization from '../../../utils/localization';
import { PanelBase } from '../../../components/panel-base/panel-base.component';
import InputField from '../../../components/field-input/field-input.component';
import { AlertMessage } from '../../../components/alert-message/alert-message.component';
import { validateURL } from '../../../validation/validation';

export const ImportLinkUpload = props => {
  const {
    importByLinkError,
    handleChangeUrl,
    handleLinkUpload,
    handleCancelImport,
    touched,
    error
  } = props;
  return (
    <PanelBase>
      <div className="mb-5">
        <h3>{localization.api.import.importApiSpecLink}</h3>
      </div>
      {importByLinkError && (
        <div className="mb-5">
          <AlertMessage type="danger">
            {localization.api.import.importApiSpecLinkError}
            <button
              className="fdk-button-small fdk-color-blue-dark"
              onClick={handleCancelImport}
            >
              {localization.api.import.toImportFile}
            </button>
          </AlertMessage>
        </div>
      )}
      <div className="mb-5">
        <InputField
          showLabel
          input={{
            name: 'importUrl',
            onChange: e => handleChangeUrl(e)
          }}
          label={localization.api.import.linkToSpec}
          type="input"
          meta={{
            touched,
            error,
            warning: null
          }}
        />
      </div>
      <Button
        testid="button-upload"
        disabled={!touched || !!error}
        className="mr-3"
        color="primary"
        onClick={handleLinkUpload}
      >
        {localization.api.import.import}
      </Button>
      <Button
        testid="button-cancel"
        className="fdk-color-blue-dark fdk-bg-color-white"
        color="transparent"
        onClick={handleCancelImport}
      >
        {localization.api.import.cancel}
      </Button>
    </PanelBase>
  );
};

ImportLinkUpload.defaultProps = {
  importByLinkError: false,
  touched: false,
  error: null
};

ImportLinkUpload.propTypes = {
  importByLinkError: PropTypes.bool,
  handleChangeUrl: PropTypes.func.isRequired,
  handleLinkUpload: PropTypes.func.isRequired,
  handleCancelImport: PropTypes.func.isRequired,
  touched: PropTypes.bool,
  error: PropTypes.string
};

const enhance = compose(
  withState('touched', 'setTouched', false),
  withState('error', 'setError', ''),
  withHandlers({
    handleChangeUrl: props => e => {
      props.handleChangeUrl(e);
      props.setTouched(true);
      const inputValue = _.get(e, ['target', 'value']);
      let errors = {};
      errors = validateURL('importUrlValidation', inputValue, errors);
      if (errors) {
        props.setError(_.get(errors, 'importUrlValidation'));
      }
    }
  })
);

export default enhance(ImportLinkUpload);
