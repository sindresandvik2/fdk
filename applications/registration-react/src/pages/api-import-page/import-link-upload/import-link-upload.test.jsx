import React from 'react';
import { shallow } from 'enzyme';
import { ImportLinkUpload } from './import-link-upload.component';

let defaultProps;
let wrapper;
let preventDefault;
let handleChangeUrl;
let handleLinkUpload;
let handleCancelImport;

beforeEach(() => {
  handleChangeUrl = jest.fn();
  handleLinkUpload = jest.fn();
  handleCancelImport = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    importByLinkError: false,
    handleChangeUrl,
    handleLinkUpload,
    handleCancelImport,
    touched: false,
    error: null
  };
  wrapper = shallow(<ImportLinkUpload {...defaultProps} />);
});

test('should render ImportLinkUpload correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle import link', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('[testid="button-upload"]').prop('onClick')(mockedEvent);
  expect(handleLinkUpload).toHaveBeenCalled();
});

test('should handle button cancel', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('[testid="button-cancel"]').prop('onClick')(mockedEvent);
  expect(handleCancelImport).toHaveBeenCalled();
});
