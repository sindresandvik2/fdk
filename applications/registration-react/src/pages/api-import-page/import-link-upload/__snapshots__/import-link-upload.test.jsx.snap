// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImportLinkUpload correctly 1`] = `
<PanelBase>
  <div
    className="mb-5"
  >
    <h3>
      Importer API-spesifikasjon via lenke
    </h3>
  </div>
  <div
    className="mb-5"
  >
    <InputField
      input={
        Object {
          "name": "importUrl",
          "onChange": [Function],
        }
      }
      label="Lenke til spesifikasjon"
      meta={
        Object {
          "error": null,
          "touched": false,
          "warning": null,
        }
      }
      showLabel={true}
      type="input"
    />
  </div>
  <Button
    className="mr-3"
    color="primary"
    disabled={true}
    onClick={[MockFunction]}
    tag="button"
    testid="button-upload"
  >
    Importer
  </Button>
  <Button
    className="fdk-color-blue-dark fdk-bg-color-white"
    color="transparent"
    onClick={[MockFunction]}
    tag="button"
    testid="button-cancel"
  >
    Avbryt
  </Button>
</PanelBase>
`;
